ladybug/legend.py

```diff
@@ -0,0 +1,877 @@
+        >> ['Slightly Cool', '', 'Neutral', '', 'Slightly Warm']
+        >> ['Cool', 'Slightly Cool', 'Neutral', 'Slightly Warm', 'Warm']
+    """
+
+    def __init__(self, values, legend_parameters=None):
+        """Initalize Ladybug Legend.
+
+        Args:
+            values: A List or Tuple of numerical values that will be used to
+                generate the legend and colors.
+            legend_parameters: An Optional LegendParameter object to override
+                default parameters of the legend.
+            """
+        # check the inputs
+        assert isinstance(values, Iterable) \
+            and not isinstance(values, (str, dict, bytes, bytearray)), \
+            'values should be a list or tuple. Got {}'.format(type(values))
+        self._values = values
+        if legend_parameters is not None:
+            assert isinstance(legend_parameters, LegendParameters), \
+                'Expected LegendParameters. Got {}.'.format(type(legend_parameters))
+            self._legend_par = legend_parameters.duplicate()
+            self._is_legend_parameters_default = False
+        else:
+            self._legend_par = LegendParameters()
+            self._is_legend_parameters_default = True
+
+        # calculate min, max and number of segments
+        self._is_min_default = False
+        self._is_max_default = False
+        if self._legend_par.min is None:
+            self._legend_par.min = min(values)
+            self._is_min_default = True
+        if self._legend_par.max is None:
+            self._legend_par.max = max(values)
+            self._is_max_default = True
+
+    @classmethod
+    def from_json(cls, data):
+        """Create a legend from a dictionary.
+
+        Args:
+            data: {
+            "values": (0, 10),
+            "legend_parameters": None}
+        """
+        if 'legend_parameters' not in data:
+            data['legend_parameters'] = None
+        legend_parameters = None
+        if data['legend_parameters'] is not None:
+            legend_parameters = LegendParameters.from_json(data['legend_parameters'])
+        for key in ('is_min_default', 'is_max_default'):
+            if key not in data:
+                data[key] = False
+
+        legend = cls(data['values'], legend_parameters)
+        legend._is_min_default = data['is_min_default']
+        legend._is_max_default = data['is_max_default']
+        return legend
+
+    @property
+    def legend_parameters(self):
+        """The legend parameters assigned to this legend."""
+        return self._legend_par
+
+    @property
+    def values(self):
+        """The data set assigned to the legend."""
+        return self._values
+
+    @property
+    def value_colors(self):
+        """A List of colors associated with the assigned values."""
+        _color_range = self.color_range
+        return tuple(_color_range.color(val) for val in self.values)
```

Comment:
Why not a list?

Reply:
You are right that there isn't much harm here in making it a list since it's always regenerated every time the property is called (so it won't create bad mutations of the object).  I guess it comes down to whether any benefits of using a list outweigh the extra memory a list has over a tuple and the fact that people will be able to make it out-of-sync with the values assigned to the object.  So what are the benefits of using a list here?

Reply:
Again since this is a very short list "the extra memory a list has over a tuple" is very small. List is easier to edit otherwise both should work. I don't know why I left this comment in the first place. This will have a very minimal effect anyways so do as you like. My bad!